lms/models/region.py

```diff
@@ -1,38 +1,25 @@
 from dataclasses import dataclass
 
 
-@dataclass(frozen=True)
```

Comment:
I don't think the `frozen=True` was necessary: neither immutability or hashabilitty of `Region`' is relevant as far as I can tell. And I have to remove it because the `__post_init__()` that I'm adding below mutates `Region`'s so they can no longer have `frozen=True`.

An alternative would be for `name` to be an `@property` then they could still be `fronzen=True`. But then constructing a `Region` with an unknown code wouldn't crash until you actually tried to read `name` (unless a `__post_init__()` with validation code were added).